ssl/quic/quic_port.c

```diff
@@ -136,9 +140,24 @@ static int port_init(QUIC_PORT *port)
     ossl_list_port_insert_tail(&port->engine->port_list, port);
     port->on_engine_list    = 1;
     port->bio_changed       = 1;
+
+    /* Generate random key for token encryption */
+    if ((ctx = EVP_CIPHER_CTX_new()) == NULL
+        || (cipher = EVP_CIPHER_fetch(port->engine->libctx,
+                                      "AES-256-GCM", NULL)) == NULL
+        || !EVP_EncryptInit_ex(ctx, cipher, NULL, NULL, NULL)
+        || (key_len = EVP_CIPHER_CTX_get_key_length(ctx)) == 0
+        || (port->token_key = OPENSSL_malloc(key_len)) == NULL
+        || !RAND_bytes_ex(port->engine->libctx, port->token_key, key_len, 0))
+        goto err;
+
+    EVP_CIPHER_CTX_free(ctx);
+    EVP_CIPHER_free(cipher);
```

Comment:
Not insisting on this, but if you made the return value a variable you could just fall into the err: label here and avoid the duplicated calls to EVP_CIPHER[_CTX]_free